docs/tutorials/demo_best_practices.py

```diff
@@ -0,0 +1,281 @@
+print(commit)
+
+# %% [markdown]
+"""
+We see that the commit has a unique id, the commit SHA which we can log to identify this particular state of the data.
+
+As the data looks good, we can also merge the branch back into `main`.
+"""
+
+# %%
+with fs.transaction as tx:
+    tx.merge(repository=REPO_NAME, source_ref=NEW_BRANCH_NAME, into="main")
+
+# %% [markdown]
+"""
+We will now start to develop our ML model. We recommend checking out a new branch for every set of experiments.
+Here, we will conduct the train test split and further experiment specific modifications, if applicable.
+"""
+
+# %%
+TRAINING_BRANCH = "experiment-1"
+with fs.transaction as tx:
+    tx.create_branch(repository=REPO_NAME, name=TRAINING_BRANCH, source="main")
+
+df = pd.read_csv(f"lakefs://{REPO_NAME}/{TRAINING_BRANCH}/weather_2010.csv")
+model_data = df.drop("time", axis=1)
+train, test = sklearn.model_selection.train_test_split(model_data, random_state=7)
+
+# %% [markdown]
+"""
+## Descriptive tags for human readability and unique SHAs for identification
+
+Since the train test split of the new branch is something we expect to address quite often in development, we will also add a human-readable tag to it.
+In the code below we directly pass the commit as the reference. If we would pass a branch, the tag would point to the current HEAD (i.e. the latest) commit on the branch.
+This makes it easy to refer back to it and to communicate this specific state of the data to colleagues. Tags are immutable which ensures consistency.
+You and your colleagues can then also work with (the same state (i.e. train test split, etc.) of the data.
+"""
+
+# %%
+TAG_NAME = "exp1_2010_data"
+with fs.transaction as tx:
+    train.to_csv(f"lakefs://{REPO_NAME}/{TRAINING_BRANCH}/train_weather.csv")
+    test.to_csv(f"lakefs://{REPO_NAME}/{TRAINING_BRANCH}/test_weather.csv")
+    commit = tx.commit(
+        repository=REPO_NAME,
+        branch=TRAINING_BRANCH,
+        message="Perform train-test split of 2010 weather data",
+    )
+    tx.tag(repository=REPO_NAME, ref=commit, tag=TAG_NAME)
+
+# %% [markdown]
+"""
+Now we have the data on different branches. If new data comes in, we can perform necessary preprocessing on a separate branch and merge it to `main` once we are sure about its compatibility and we have run all the necessary tests.
+Should the new data be important for the experimentation as well, then we can merge the new main branch into the experimentation branch.
+Then we can create a new tag for the dataset.
+Tags cannot be directly reassigned. If you want to do this anyways, for example to prevent namespace clutter, you have to delete the tag and create a new one.
+However, beware as this might break reproducibility in other places (i.e. colleagues might expect unchanged data).
+To ensure failsafe versioning use the SHA's of the commits instead of tags or branch names in experiment tracking tools.
+"""
```

Comment:
This section is quite hard to read. At the very least, I would split it into multiple paragraphs.